10up/ElasticPress
# Feature/1932

### Description of the Change

Display a warning notice when choosing to enable a feature when that feature requires reindexing. Then, when the user clicks to save the settings, warn the user again and ask them to confirm that a reindex will occur. 

### Benefits

Lets a user know, and asks them to confirm, that a lengthy reindex could occur when enabling a feature that requires a reindex.

### Verification Process

1. Go to a setting that requires reindexing. Select "Enabled" a warning message should appear.
2. Click to disable the setting, the warning message should disappear.
3. Click to enable it again and the warning will reappear.
4. Click save, a browser confirm dialog should ask the for confirmation.
5. Press cancel, setting should not save.
6. Click save again, and OK in the confirm dialog. Settings will save and you will be redirected to the sync page.
7. Go back to Features, choose to disable the feature and save. There should be no warning or confirmation. No redirect should occur.
8. After saving is complete, click to Enable the feature. The warning should appear again.
9. Click to save, the confirmation should appear again.

### Checklist:






### Applicable Issues

Resolves #1932.

### Changelog Entry

- Display a notice that content will be reindexed when enabling a feature that requires it.
